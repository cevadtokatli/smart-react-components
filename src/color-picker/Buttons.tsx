import React from "react"
import styled from "styled-components"
import size from "../constants/color-picker/size"
import {Language} from "../types/color-picker"
import Button from "../button/Button"
import SaveIcon from "./icons/Save"
import CancelIcon from "./icons/Cancel"

const ButtonContainer = styled.div`
    display: flex;
    padding-top: ${size.space.default}px;
    margin: -${size.space.small}px;

    > a {
        flex: 1 1 calc(50% - ${size.space.small*2}px);
        justify-content: center;
        margin: ${size.space.small}px;
        white-space: nowrap;

        svg {
            flex: 0 0 auto;
            width: ${size.icon.width}px;
            height: ${size.icon.height}px;
            margin-right: ${size.space.small}px;
        }
    }
`

interface Props {
    type: string
    save: () => void
    cancel: () => void
    lang: Language
}

const Buttons: React.FC<Props> = ({type,save,cancel,lang}) => {
    return (
        <ButtonContainer>
            <Button type={type} size="small" elementProps={{onClick:save}}>
                <SaveIcon />
                {lang.save}
            </Button>
            <Button type={type} size="small" elementProps={{onClick:cancel}}>
                <CancelIcon />
                {lang.cancel}
            </Button>
        </ButtonContainer>
    )
}

export default Buttons