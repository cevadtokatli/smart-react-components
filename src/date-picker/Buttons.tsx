import React from "react"
import {Value, Language} from "../types/date-picker"
import {SetValue} from "../props"
import Div from "../element/Div"
import Button from "../button/Button"

interface Props {
    value: Value
    setValue: SetValue<Value>
    initialValue: Value
    setInitialValue: React.Dispatch<React.SetStateAction<Value>>
    onSave: () => void
    onCancel: () => void
    type: string
    lang: Language
}

const Buttons: React.FC<Props> = ({value,setValue,initialValue,setInitialValue,onSave,onCancel,type,lang}) => {
    /**
     * Sets initial value as value.
     */
    const cancel = () => {
        setValue(initialValue)
        if(onCancel)
            onCancel()
    }

    /**
     * Sets value as initial value.
     */
    const save = () => {
        setInitialValue(value)
        if(onSave)
            onSave()
    }

    return (
        <Div className="src-date-picker-buttons" display="flex" justifyContent="flex-end" alignItems="center">
            <Button type={type} link={true} waveEffect="dark" elementProps={{onClick:cancel}}>{lang.cancel}</Button>
            <Button type={type} link={true} waveEffect="dark" elementProps={{onClick:save}}>{lang.save}</Button>
        </Div>
    )
}

export default Buttons