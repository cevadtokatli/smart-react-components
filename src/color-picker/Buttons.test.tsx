import React from "react"
import renderTheme from "../test/renderTheme"
import mountTheme from "../test/mountTheme"
import DEFAULT_LANGUAGE from "../constants/color-picker/default-language"
import Buttons from "./Buttons"

describe("<Buttons />", () => {
    let props

    beforeEach(() => {
        props = {
            type: "primary",
            save: jest.fn(),
            cancel: jest.fn(),
            lang: DEFAULT_LANGUAGE
        }
    })

    it("should match snapshot", () => {
        const wrapper = renderTheme(<Buttons {...props} />)
        expect(wrapper).toMatchSnapshot()
    })

    it("should call save method", () => {
        const wrapper = mountTheme(<Buttons {...props} />)
        wrapper.find("a").at(0).simulate("click")
        expect(props.save).toHaveBeenCalled()
    })

    it("should call cancel method", () => {
        const wrapper = mountTheme(<Buttons {...props} />)
        wrapper.find("a").at(1).simulate("click")
        expect(props.cancel).toHaveBeenCalled()
    })
})