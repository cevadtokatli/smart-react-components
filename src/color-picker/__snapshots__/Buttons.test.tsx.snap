// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Buttons /> should match snapshot 1`] = `
<div
  class="sc-Axmtr fFPKYE"
>
  <a
    class="sc-AxirZ sc-AxiKw bvfXbn"
    data-src-wave-effect="0"
    shape="default"
    tabindex="0"
  >
    <svg
      class="sc-AxhUy sc-AxgMl ewSVZA"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
      />
    </svg>
    Save
  </a>
  <a
    class="sc-AxirZ sc-AxiKw bvfXbn"
    data-src-wave-effect="0"
    shape="default"
    tabindex="0"
  >
    <svg
      class="sc-AxhUy sc-AxheI dVkpdy"
      viewBox="0 0 1792 1792"
    >
      <path
        d="M1490 1322q0 40-28 68l-136 136q-28 28-68 28t-68-28l-294-294-294 294q-28 28-68 28t-68-28l-136-136q-28-28-28-68t28-68l294-294-294-294q-28-28-28-68t28-68l136-136q28-28 68-28t68 28l294 294 294-294q28-28 68-28t68 28l136 136q28 28 28 68t-28 68l-294 294 294 294q28 28 28 68z"
      />
    </svg>
    Cancel
  </a>
</div>
`;
